idunn/blocks/images.py

```diff
@@ -0,0 +1,131 @@
+        displayErrorImage = False
+
+        salt = cls.get_salt()
+        token = f"{source}{width}x{height}{salt}"
+        hash = hashlib.sha256(bytes(token, encoding='utf8')).hexdigest()
+        domain = cls.get_domain(hash)
+
+        size = int(width) * int(height)
+
+        hashURLpart = f"{hash[0]}/{hash[1]}/{hash[2:]}"
+
+        filename = source.split('/')[-1]
+        if not bool(re.match("^.*\.(jpg|jpeg|png|gif)$", filename)):
+            filename += ".jpg"
+
+        params = f"u={urllib.parse.quote(source)}&q={1 if displayErrorImage else 0}&b={1 if bestFit else 0}&p={1 if progressive else 0}&a={1 if animated else 0}"
+        return domain + "/" + str(size) + "/" + hashURLpart + "/" + filename + "?" + params
+
+class Image(types.Type):
+    url = validators.String()
+    alt = validators.String()
+    credits = validators.String()
+
+class ImagesBlock(BaseBlock):
+    BLOCK_TYPE = "images"
+    images = validators.Array(items=Image)
+
+    @classmethod
+    def from_es(cls, es_poi, lang):
+        wikidata_id = es_poi.get("properties", {}).get("wikidata")
+        if wikidata_id is not None:
+            wiki_index = WikidataConnector.get_wiki_index(lang)
+            if wiki_index is not None:
+                properties = es_poi.get('properties', {})
+                place_name = properties.get('name')
+                try:
+                    try:
+                        wiki_es = WikidataConnector.get_wiki_es()
+                    except WikiUndefinedException:
+                        logger.info("WIKI_ES variable has not been set: we cannot return any image")
+                        return None
+                    if wiki_es is None:
+                        return None
+                    resp = wiki_es.search(
+                        index=wiki_index,
+                        body={
+                            "filter": {
+                                "term": {
+                                    "wikibase_item": wikidata_id
+                                }
+                            }
+                        }
+                    ).get('hits', {}).get('hits', [])
+                except ConnectionError:
+                    logger.warning("Wiki ES not available: connection exception raised", exc_info=True)
+                    return None
+                if len(resp) == 0:
+                    return None
+
+                raw_url= resp[0]['_source'].get('pageimage_thumb')
+                if raw_url is None:
+                    return None
+
+                images = None
+                if not ThumbrHelper.use_thumbr():
+                    image_wiki = Image(url=urllib.parse.quote(raw_url), alt=place_name, credits="wikimedia")
+                    images = [image_wiki]
+                else:
+                    thumbr_url_large = ThumbrHelper.get_url_remote_thumbnail(
+                        source=raw_url,
+                        width=30,
+                        height=30
```

Comment:
we need to think about how to set these values.  
I suppose the ratio of images is not fixed. Is the size of the image present in the response ?  
Should we force a ratio on our side, or set these values dynamically ?  
To discuss with the frontend ; it's also possible to let the frontend set the `bestFit` value to truncate the image if needed.